src/transformers/models/mobilevit/modeling_mobilevit.py

```diff
@@ -0,0 +1,1098 @@
+        return hidden_states
+
+
+class MobileViTTransformerLayer(nn.Module):
+    def __init__(self, config: MobileViTConfig, hidden_size: int, intermediate_size: int) -> None:
+        super().__init__()
+        self.attention = MobileViTAttention(config, hidden_size)
+        self.intermediate = MobileViTIntermediate(config, hidden_size, intermediate_size)
+        self.output = MobileViTOutput(config, hidden_size, intermediate_size)
+        self.layernorm_before = nn.LayerNorm(hidden_size, eps=config.layer_norm_eps)
+        self.layernorm_after = nn.LayerNorm(hidden_size, eps=config.layer_norm_eps)
+
+    def forward(self, hidden_states: torch.Tensor) -> torch.Tensor:
+        attention_output = self.attention(self.layernorm_before(hidden_states))
+        hidden_states = attention_output + hidden_states
+
+        layer_output = self.layernorm_after(hidden_states)
+        layer_output = self.intermediate(layer_output)
+        layer_output = self.output(layer_output, hidden_states)
+        return layer_output
+
+
+class MobileViTLayer(nn.Module):
+    """
+    MobileViT block: https://arxiv.org/abs/2110.02178
+    """
+
+    def __init__(
+        self,
+        config: MobileViTConfig,
+        in_channels: int,
+        hidden_size: int,
+        num_stages: int,
+    ) -> None:
+        super().__init__()
+        self.patch_w = config.patch_size
+        self.patch_h = config.patch_size
+
+        self.conv_kxk = MobileViTConvLayer(
+            config,
+            in_channels=in_channels,
+            out_channels=in_channels,
+            kernel_size=config.conv_kernel_size,
+        )
+
+        self.conv_1x1 = MobileViTConvLayer(
+            config,
+            in_channels=in_channels,
+            out_channels=hidden_size,
+            kernel_size=1,
+            use_norm=False,
+            use_act=False,
+        )
+
+        self.transformer = nn.Sequential()
+        for block_idx in range(num_stages):
+            transformer = MobileViTTransformerLayer(
+                config,
+                hidden_size=hidden_size,
+                intermediate_size=int(hidden_size * config.mlp_ratio),
+            )
+            self.transformer.append(transformer)
+
+        self.layernorm = nn.LayerNorm(hidden_size, eps=config.layer_norm_eps)
+
+        self.conv_proj = MobileViTConvLayer(
+            config,
+            in_channels=hidden_size,
+            out_channels=in_channels,
+            kernel_size=1,
+        )
+
+        self.fusion = MobileViTConvLayer(
+            config,
+            in_channels=2 * in_channels,
+            out_channels=in_channels,
+            kernel_size=config.conv_kernel_size,
+        )
+
+    def unfolding(self, features: torch.Tensor) -> Tuple[torch.Tensor, Dict]:
+        patch_w, patch_h = self.patch_w, self.patch_h
+        patch_area = int(patch_w * patch_h)  # P
+
+        batch_size, channels, orig_h, orig_w = features.shape  # B, C, H, W
```

Comment:
```suggestion
        batch_size, channels, orig_h, orig_w = features.shape
```
The comment is not helpful, the variables are properly named already :-)
I'd advocate to do h->height and w->width in all this function though.

Reply:
fixed